src/etos_suite_runner/lib/runner.py

```diff
@@ -49,33 +53,97 @@ def __init__(self, params, etos, context):
 
         self.context = context
 
-    def _run_etr_and_wait(self, environment):
-        """Run ETR based on number of IUTs and wait for them to finish.
+    def _release_environment(self, task_id):
+        """Release an environment from the environment provider.
+
+        :param task_id: Task ID to release.
+        :type task_id: str
+        """
+        wait_generator = self.etos.http.wait_for_request(
+            self.etos.debug.environment_provider, params={"release": task_id}
+        )
+        for response in wait_generator:
+            if response:
+                break
+
+    def _run_etr(self, environment):
+        """Trigger an instance of ETR.
 
         :param environment: Environment which to execute in.
         :type environment: dict
-        :return: List of test results from all ETR instances.
-        :rtype: list
         """
+        uri = environment["data"]["uri"]
+        json_header = {"Accept": "application/json"}
+        json_response = self.etos.http.wait_for_request(
+            uri,
+            headers=json_header,
+        )
+        suite = {}
+        for suite in json_response:
+            break
+        else:
+            raise Exception("Could not download sub suite instructions")
+
+        executor = Executor(self.etos)
+        executor.run_tests(suite)
+
+    def _environments(self, suite_name):
```

Comment:
If there is a TERCC with two suites in it and one of them succeeds their checkout and the other fails, then there is a risk that the first suite has already started up and we will be checking in the environment before that first one has finished using it.
We must make sure that all suites that have started finish, before we attempt to check in the whole environment.